File: 1688. Count of Matches in Tournament/Solution.java
class Solution {
    public int numberOfMatches(int n) {
        return nextMatch(n,0);
    }

    private int nextMatch(int n, int tot){
        if(n < 2) return tot;
        int res = n/2;
        if(n%2 != 0) return nextMatch(res + 1, tot + res);
        else return nextMatch(res, tot + res);        
    }
}

File: 1688. Count of Matches in Tournament/Solution.py
class Solution:
    def nextMatch(self,n : int,total : int) -> int: 
        if n < 2 :
            return total
        
        res = n//2
        
        if n % 2 != 0:
            return self.nextMatch(res+1,total+res)
        else:
            return self.nextMatch(res,total+res)
        
    def numberOfMatches(self, n: int) -> int:
        return self.nextMatch(n,0)